### PR TITLE
Feature: Implement OR filtering

### PR DESCRIPTION
**Pull request by perfectmak**
Change log
- Refactor to Allow AND and OR filters combination
- Allow overriding query for columns to handle edge cases.
- Updates documentation

### CODE REVIEW

File: tests/knex-flex-filter.test.js [hunk 1]
```diff
@@ -243,6 +243,21 @@ describe('knex-flex-filter', () => {
...
+        knex.table('entities'),
+        {
+          ownerId_eq: 1,
+          or_name_contains: 'Peter Jackson',
+        },
+        { castFn },
+      );
+
+      const result = await query;
+      expect(parseInt(result[0].ownerId, 10)).toEqual(1);
```
Comment (dmerrill6) [inline tests/knex-flex-filter.test.js]: What do you think of adding an additional `expect` that validates that the sql generated is the correct one with the `or`?

**Comment by perfectmak** [comment]
Do not merge this yet. I'm update the same branch to handle the `OR` and `AND` filtering we mention.

File: src/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-use-before-define */
```
Comment (perfectmak) [inline src/index.js]: recursive call between `parseFilter` and `parseGroupFilter`
